api/controllers/HistoryController.js

```diff
@@ -59,17 +59,23 @@ module.exports = {
 
     if (!machineId) {
       return res.badRequest('Invalid machine id');
+    } else if (!req.body.data.attributes.endDate
+    || req.body.data.attributes.endDate === '') {
```

Comment:
If `req.body.data.attributes.endDate === ''` then `!req.body.data.attributes.endDate === true`.

Replace with `if (req.body.data.attributes.endDate === '')` is enough.